.github/workflows/all-in-one-build.yml

```diff
@@ -0,0 +1,35 @@
+name: Docker
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  all-in-one:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: true
+
+    - uses: actions/setup-go@v2
+      with:
+        go-version: ^1.15
+
+    - uses: docker/login-action@v1
+      with:
+        username: ${{ secrets.DOCKERHUB_USERNAME }}
```

Comment:
Would it make more sense not to encrypt the user name, which happens to be `jaegertracingbot`?

Reply:
Yes I do agree.

Reply:
I defined the token in secrets
![image](https://user-images.githubusercontent.com/3523016/100495631-5e3f0b00-311b-11eb-8eb2-44181a07b22b.png)

Reply:
Thank you.

Reply:
Not sure if GitHub Actions can see secrets at organisation level.

Reply:
Hard to imagine why it wouldn't. The token itself is working, I just tested it
```
$ docker login
Login with your Docker ID to push and pull images from Docker Hub. If you don't have a Docker ID, head over to https://hub.docker.com to create one.
Username: jaegertracingbot
Password:
Login Succeeded
```

Reply:
The recent run failed https://github.com/jaegertracing/jaeger/pull/2663/checks?check_run_id=1466696346

Reply:
The Settings page says:

> Secrets are not passed to workflows that are triggered by a pull request from a fork.

This is why I was warning about TRAVIS_SECURE_ENV_VARS. The previous setup was intentionally organized in such a way that almost all of the CI job is executed, and only the push to the registry is skipped when it's not from master.

Reply:
We might need to define the secret at repo level instead of org level.

Reply:
Ah I see.

Reply:
Finally able to make it work by doing docker login only if the secret is defined. Not sure if this is okay method.

Reply:
I think this will leak secret so definitely not the way.

Reply:
I tested in my jaeger fork putting a random string for DOCKERHUB_TOKEN. It should be all fine. I was worried about leaking it in logs but looks like encrypted secrets never gets printed. I tested it. Also their documentation says to use secrets to populate env vars, so it should be all good (https://docs.github.com/en/free-pro-team@latest/actions/reference/encrypted-secrets#using-encrypted-secrets-in-a-workflow)

<img width="1285" alt="docker-registry" src="https://user-images.githubusercontent.com/16268376/100514605-2a162980-31b1-11eb-89cd-6d91f475bd5f.png">